Sources/StreamChat/Repositories/MessageRepository.swift

```diff
@@ -88,6 +88,47 @@ class MessageRepository {
             })
         }
     }
+    
+    /// Marks the message's local status to failed and adds it to the offline retry which sends the message when connection comes back.
+    func scheduleOfflineRetry(for messageId: MessageId, completion: @escaping (Result<ChatMessage, MessageRepositoryError>) -> Void) {
```

Comment:
MessageSender uses this to send queued messages directly to offline repo when it sees that there is connection issue.